plugin/pkg/global/resourcereferencemanager/admission.go

```diff
@@ -865,10 +866,54 @@ func (r *ReferenceManager) ensureBindingReferences(ctx context.Context, attribut
 }
 
 func (r *ReferenceManager) ensureShootReferences(ctx context.Context, attributes admission.Attributes, oldShoot, shoot *core.Shoot) error {
-	if utils.BuildCloudProfileReference(shoot) != utils.BuildCloudProfileReference(oldShoot) {
-		if _, err := utils.GetCloudProfileSpec(r.cloudProfileLister, r.namespacedCloudProfileLister, shoot); err != nil {
+	newCloudProfileReference := utils.BuildCloudProfileReference(shoot)
+	oldCloudProfileReference := utils.BuildCloudProfileReference(oldShoot)
+	if !apiequality.Semantic.DeepEqual(newCloudProfileReference, oldCloudProfileReference) {
+		newCloudProfileSpec, err := utils.GetCloudProfileSpec(r.cloudProfileLister, r.namespacedCloudProfileLister, shoot)
+		if err != nil {
 			return fmt.Errorf("could not find cloudProfileSpec from the shoot cloudProfile reference: %s", err.Error())
 		}
+		if oldCloudProfileReference != nil &&
+			(oldCloudProfileReference.Kind != newCloudProfileReference.Kind || oldCloudProfileReference.Name != newCloudProfileReference.Name) {
+			newCloudProfileSpecCore := &core.CloudProfileSpec{}
+			if err := api.Scheme.Convert(newCloudProfileSpec, newCloudProfileSpecCore, nil); err != nil {
+				return err
+			}
+			oldCloudProfileSpec, err := utils.GetCloudProfileSpec(r.cloudProfileLister, r.namespacedCloudProfileLister, oldShoot)
+			if err != nil {
+				return fmt.Errorf("could not find cloudProfileSpec from the shoot cloudProfile reference: %s", err.Error())
+			}
+			oldCloudProfileSpecCore := &core.CloudProfileSpec{}
+			if err := api.Scheme.Convert(oldCloudProfileSpec, oldCloudProfileSpecCore, nil); err != nil {
+				return err
+			}
+
+			// Check that the target cloud profile still supports the currently used machine types, machine images and volume types.
+			// No need to check for Kubernetes versions, as the NamespacedCloudProfile could have only extended a version so with the next maintenance the Shoot will be updated to a supported version.
```

Comment:
The assumed change is from `NamespacedCloudProfile` to `CloudProfile`, but it isn't checked that they are really related, is it?

I only see a content-wise check but it can happen a change to a completely different `CloudProfile` happens this way, and thus the Kubernetes version assumption is not fulfilled anymore.

Could you please double check?

Reply:
Changes to a completely different `CloudProfile` are prohibited by a check in the shoot validator (calling `#ValidateCloudProfileChanges(...)`), which asserts the correct hierarchical relation. Hence my assumption that for Kubernetes versions, only expiration dates can change.